game/src/Game/GameSession/billboard.gdshader

```diff
@@ -0,0 +1,45 @@
+shader_type spatial;
+//depth_test_disabled along with the sorting offset of 10
+//ensures this will be drawn in front of other objects in 
+//3d space.
+render_mode unshaded, depth_test_disabled;
+
+
+//vic2 only ever loads a max of 12 BillboardType
+//Of these, only 4 are actually used
+uniform sampler2D billboards[12] : source_color;
+uniform int numframes[12];
+uniform float sizes[12];
+
+uniform float height_add_factor = 0.3;
+
+//COLOR/INSTANCE_CUSTOM is our custom data, used as follows:
+// x=image index
+// y=frame in image index
+// z,w are unused for now
+
+void vertex() {
+	COLOR = INSTANCE_CUSTOM; //send instance_custom info to fragment
+	float size = sizes[int(COLOR.x)];
+	
+	vec3 cam_right_worldspace = vec3(VIEW_MATRIX[0][0],VIEW_MATRIX[1][0],VIEW_MATRIX[2][0]);
+	vec3 cam_up_worldspace = vec3(VIEW_MATRIX[0][1],VIEW_MATRIX[1][1],VIEW_MATRIX[2][1]);
+	VERTEX =
+		cam_right_worldspace * VERTEX.x +
+		cam_up_worldspace * VERTEX.y;
```

Comment:
```suggestion
	VERTEX = (vec4(VERTEX, 1.0) * VIEW_MATRIX).xyz;
```
This can be reduced to a matrix multiplication, which on a GPU should be faster than the separate operations it's replacing. Although currently providing `0.0` and `1.0` for the `w` coordinate produces the same results, it's better to go with `1.0` for future-proofing as it's needed for translation transforms to work - `(x, y, z, 1)` represents a position which will be moved by translations, while `(x, y, z, 0)` represents a direction and so is invariant under translations.

Reply:
Works.